src/python/pants/backend/jvm/tasks/jvm_compile/class_not_found_error_patterns.py

```diff
@@ -35,6 +35,26 @@
    'is not a member of package (\S+)\n'
    '\s*\[error\] import (?P<classname>\S+)'),
   (r'\s*\[error\] Class (?P<classname>\S+) not found \- continuing with a stub\.'),
+  
+  # scalac errors with strict_deps enabled. Some errors may still be missing.
+  # below matches some of the common issues caused by missing exports.
+  (r'\s*\[error\] [^:]+\.scala:\d+:\d+: Symbol \'\S+ \<none\>\.(?P<classname>\S+)\' is missing from the '
+   r'classpath\.\n\s*\[error\] This symbol is required by \'(?P<type>\S+) (?P<dependee_classname>\S+)\'\.\n'
+   r'\s*\[error\] Make sure that \S+ (?P<classnameonly>\S+) is in your classpath and check for conflicting '
+   r'dependencies with `-Ylog-classpath`\.\n\s*\[error\] A full rebuild may help if '
+   r'\'(?P<dependee_classnameonly>\S+)\.class\' was compiled against an incompatible version of '
+   r'\<none\>\.(?P<packagename>\S+)\.'),
+  (r'\s*\[error\] [^:]+\.scala:\d+:\d+: Symbol \'\S+ (?P<classname>\S+)\' is missing from the classpath\.\n\s*'
+   r'\[error\] This symbol is required by \'(?P<type>\S+) (?P<dependee_classname>\S+)\'\.\n\s*'
+   r'\[error\] Make sure that \S+ (?P<classnameonly>\S+) is in your classpath and check for conflicting dependencies '
+   r'with `-Ylog-classpath`\.\n\s*\[error\] A full rebuild may help if \'(?P<dependee_classnameonly>\S+)\.class\' '
+   r'was compiled against an incompatible version of (?P<packagename>\S+)\.'),
+  # covers member types of traits not used in the extending type.
+  (r'\s*\[error\] Symbol \'\S+ (?P<classname>\S+)\' is missing from the classpath\.\n\s*\[error\] This symbol is '
+   r'required by \'method (?P<dependee_classname>\S+)\.(?P<method_name>\S+)\'\.\n\s*\[error\] Make sure that \S+ '
```

Comment:
I think it'd be good to specifically exclude periods from the method name group for this matcher.

It should work without it because the dependee_classname will greedily consume the `.`s before the last `.`, but I think the pattern would be clearer that way.